### DISCUSSION TITLE
Is there any data collected by the pendo sdk?

### DISCUSSION THREAD

**Discussion by stremsdoerfer** [pendo-io/pendo-mobile-sdk#90] [General]
Hi !

In order for us to provide accurate Privacy Nutrition Labels to Apple during our review, we need to know whether all our third party libraries collect any kind of data. Is there any data collected by the Pendo sdk that we should specify in our review? (such as crash, usage, diagnostics etc.)

Thanks!

**Comment by MikePendo** [comment]
@stremsdoerfer 
Sorry for not replying for a long time as we dont have 100% clear answer about all of that yet 
So in terms of API: [Describing use of required reason API](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_use_of_required_reason_api?language=objc)
Starting from ~~3.0.1~~ 3.0.0 we will remove all the left overs of those apis
Regarding  the [Describing data use in privacy manifests](https://developer.apple.com/documentation/bundleresources/privacy_manifest_files/describing_data_use_in_privacy_manifests?language=objc) Things are a little bit more trickier here as the SDK scan the screen and hash truncate every string (basically they are used for screen identification in some rare case). So we might collect a hash of some email implicitly (please note that we dont collect text inputs from uitextfields/uitextviews). Obviously if you supply a visitorId to PendoSDK in form of your email that will be explicitly collected. In any case the manifest should be supplied by the SDK in that case. So we will need to further debate it internally and see if we need such a manifest.
Regarding your concrete question such as crash/usage/diagnostics - NO, we dont collect any of those

**Reply by MikePendo** [reply]
@stremsdoerfer 
We have just released our native SDK with `.xcprivacy`
